lib/new_relic/agent/pipe_channel_manager.rb

```diff
@@ -247,6 +292,11 @@ def unmarshal(data)
         rescue StandardError => e
           ::NewRelic::Agent.logger.error "Failure unmarshalling message from Resque child process, pid #{Process.pid}, size #{data.size}", e
           ::NewRelic::Agent.logger.error Base64.encode64(data)
+          if nr_debug? && File.exist?('/tmp/marshall_log')
+            content = File.binread('/tmp/marshall_log')
+            ::NewRelic::Agent.logger.error "Sent #{content.size}"
+            ::NewRelic::Agent.logger.error Base64.encode64(content)
```

Comment:
These files might get very very long, I'm not 100% sure.  Maybe we should cap it off at 100 MiB or something?  If it's too big, we won't dump the data.  I think we don't want to screw up staging either.

Reply:
One file per job. I tested it.